unit/src/DependencyResolverTest.C

```diff
@@ -144,3 +144,21 @@ TEST_F(DependencyResolverTest, dependsOnTest)
   EXPECT_FALSE(_tree.dependsOn("m1", "something_else"));
   EXPECT_FALSE(_tree.dependsOn("something_else", "k0"));
 }
+
+TEST_F(DependencyResolverTest, cyclicTest)
+{
+  try
+  {
+    // Attempt to insert a dependency that results in cyclicity
+    _resolver.insertDependency("a", "b");
+    FAIL() << "missing expected exception";
+  }
+  catch (const std::exception & e)
```

Comment:
I think i have to throw a CyclicDependencyException in DependencyResolver.h.  Currently i'm just using mooseError.  The mooseError is caught by

```
catch (const std::exception & e)
```

however the following doesn't catch the mooseError;

```
 catch (CyclicDependencyException<std::shared_ptr<std::string>> & e)
 ```
Do you agree, @permcody ?